src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -41,14 +41,25 @@ internal class RuntimeBroker : IBroker
 
                 return new NativeInterop.Core();
             }
-            catch (Exception ex)
+            catch (MsalRuntimeException ex) when (ex.Status == ResponseStatus.ApiContractViolation)
             {
+                // failed to initialize msal runtime - can happen on older versions of Windows. Means broker is not available.
+                // We will never get here with our current OS version check. Instead in this scenario we will fallback to the browser
+                // but MSALRuntime does it's internal check for OS compatibility and throws an ApiContractViolation MsalRuntimeException.
+                // For any reason, if our OS check fails then this will catch the MsalRuntimeException and 
+                // log but we will not fallback to the browser in this case. 
                 s_initException = ex;
 
                 // ignored
                 return null;
             }
-
+            catch (Exception ex)
+            {
+                // When MSAL Runtime dlls fails to load then we catch the exception and throw with a meaningful
+                // message with information on how to troubleshoot
+                throw new MsalClientException(
+                    "wam_runtime_init_failed", ex.Message + " See https://aka.ms/msal-net-wam#troubleshooting");
```

Comment:
`ex` may need to be included as an inner exception.

Reply:
added